### PR TITLE
fix: fetch composio apps list dynamically

### PR DESCRIPTION
**Pull request by tushar-composio**
Composio Apps list is only loaded after the toolset is initialized. This change ensures that the toolset is initialized before the app options are populated.

### CODE REVIEW

File: src/backend/base/langflow/components/composio/composio_api.py [hunk 1]
```diff
@@ -223,6 +223,9 @@ def _get_normalized_app_name(self) -> str:
 
     def update_build_config(self, build_config: dict, field_value: Any, field_name: str | None = None) -> dict:  # noqa: ARG002
         # First, ensure all dynamic fields are hidden by default
+        toolset = self._build_wrapper()
+        build_config["app_names"]["options"] = list(App.iter())
```
Comment (tushar-composio) [inline src/backend/base/langflow/components/composio/composio_api.py:227]:
```suggestion
        if hasattr(self, "app_names") and self.app_names != "":
            toolset = self._build_wrapper()
            build_config["app_names"]["options"] = list(App.iter())
```

File: src/backend/base/langflow/components/composio/composio_api.py [hunk 2]
```diff
@@ -223,6 +223,10 @@ def _get_normalized_app_name(self) -> str:
 
     def update_build_config(self, build_config: dict, field_value: Any, field_name: str | None = None) -> dict:  # noqa: ARG002
         # First, ensure all dynamic fields are hidden by default
+        if hasattr(self, "app_names") and self.app_names != "":
+            toolset = self._build_wrapper()
+            build_config["app_names"]["options"] = list(App.iter())
+
```
Comment (tushar-composio) [inline src/backend/base/langflow/components/composio/composio_api.py:229]:
```suggestion
        # Update the available apps options from the API
        if hasattr(self, "app_names") and self.app_names != "":
            toolset = self._build_wrapper()
            build_config["app_names"]["options"] = list(App.iter())

        # First, ensure all dynamic fields are hidden by default
```

File: src/backend/base/langflow/components/composio/composio_api.py (hunk 1 above)
Comment (abhishekpatil4) [reply src/backend/base/langflow/components/composio/composio_api.py]: Shouldn't it be? (ln 226)
```
if self.api_key:
            toolset = self._build_wrapper()
            build_config["app_names"]["options"] = list(App.iter())
```

File: src/backend/base/langflow/components/composio/composio_api.py [hunk 3]
```diff
@@ -222,6 +222,11 @@ def _get_normalized_app_name(self) -> str:
         return self.app_names.replace(" ✅", "").replace("_connected", "")
 
     def update_build_config(self, build_config: dict, field_value: Any, field_name: str | None = None) -> dict:  # noqa: ARG002
+        # Update the available apps options from the API
+        if hasattr(self, "app_names") and self.app_names != "":
```
Comment (tushar-composio) [inline src/backend/base/langflow/components/composio/composio_api.py:226]:
```suggestion
        if hasattr(self, "api_key") and self.api_key != "":
```

**Review by ogabrielluiz — APPROVED** [review]
Nice, @tushar-composio . 
Thank you! 

LGTM

**Comment by tusharsadhwani** [comment]
@ogabrielluiz codeflash seems to not be working right now, so automerge won't merge the PR.

**Pull request merged**